industry_benchmarks/utils/traj_cleanup.py

```diff
@@ -0,0 +1,74 @@
+import pathlib
+from openmmtools import multistate
+import numpy as np
+import MDAnalysis as mda
+from openfe_analysis import FEReader
+
+
+def compute_mbar_energies(analyzer):
+    """
+    Returns
+    ------
+    u_ln: energy matrix of shape (K,N) indexed by k,n
+        K is the total number of states observables are desired.
+        N is the total number of samples drawn from ALL states.
+        The nth configuration is the energy evaluated in the kth thermodynamic state.
+    N_l: 1-D iterable of shape K
+        The number of samples drawn from each kth state.
+    """
+    analyzer.use_full_trajectory = True
+    u_ln, N_l = analyzer._compute_mbar_decorrelated_energies()
+    return u_ln, N_l
+
+
+def get_replica_state_indices(analyzer):
+    energy_data = list(analyzer._read_energies(truncate_max_n_iterations=True))
+    replicas_state_indices = energy_data[-1]
+    return replicas_state_indices
+
+
+def subsample_traj(simulation, hybrid_system_pdb, lambda_windows, outfile):
+    for i in range(0, lambda_windows):
+        u = mda.Universe(hybrid_system_pdb, simulation,
+                         format=FEReader, state_id=i)
+        skip = round(len(u.trajectory) / 20)
+        with mda.Writer(f'{outfile}_{i}.xtc', n_atoms=len(u.atoms)) as w:
```

Comment:
This seems like it's expecting a str rather than a path. We'll be calling this script from someone far from where the files are & where they need to be written, so it's worth testing that it'll hold up if you read from remote directory and write to remote directory.

Reply:
I changed this to instead take in the string directly from the pathlib.Path, hopefully that is more stable?

Reply:
For internal use it'll do I think.